### PR TITLE
Exercise model validations, changing exercise form to simple form, adding errors when a user puts a wrong input into the exercise form, and adding guard-rspec and shoulda-matchers for testing

### PR DESCRIPTION
**Pull request by josephferrairo**
I added alerts for exercises with bad inputs using the bootstrap simple form on the front end, added simple form bootstrap, and added a model validation for exercises which makes the reps have to be greater than or equal to 1

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ default: &default
   # http://guides.rubyonrails.org/configuring.html#database-pooling
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
 
+
```
Comment (BGRicker) [inline config/database.yml]: blank line

File: app/controllers/exercises_controller.rb [hunk 2]
```diff
@@ -1,13 +1,20 @@
 class ExercisesController < ApplicationController
 
   def new
+    @workout = Workout.find(params[:workout_id])
```
Comment (BGRicker) [inline app/controllers/exercises_controller.rb]: this wasn't needed before, why is it needed now

File: app/controllers/exercises_controller.rb [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
     @workout = Workout.find(params[:workout_id])
-    @workout.exercises.create(exercise_params)
-    redirect_to @workout
+    @exercise = @workout.exercises.create(exercise_params)
+    if @exercise.save
+      flash[:success] = 'New Workout has been created!'
+      redirect_to workout_path(@workout)
+    else
+      render :_new , status:  :unprocessable_entity
+    # currently this renders only the new page, I would like to make it so that it renders the entire view of this page when an error is present
```
Comment (BGRicker) [inline app/controllers/exercises_controller.rb]: I don't think you need to check in comments

File: app/views/exercises/_new.html.erb [hunk 4]
```diff
@@ -1,31 +1,14 @@
...
-	  <% end %>
+  <%= simple_form_for([@workout, @exercise], html: {class: 'form-horizontal'}) do |f| %>
+    <%= f.input :name, label: 'Name' %>
+    <%= f.input :reps, label: 'Reps' %>
+    <%= f.input :note, :as => :text, label: 'Notes' %>
+    <%= f.submit "Submit" %>
+  <% end %>
 </div>
 
+
```
Comment (BGRicker) [inline app/views/exercises/_new.html.erb]: blank line

File: config/initializers/simple_form_bootstrap.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
+# Use this setup block to configure all options available in SimpleForm.
```
Comment (BGRicker) [inline config/initializers/simple_form_bootstrap.rb]: is simple form bootstrap just for styling?

**Comment by BGRicker** [comment]
When you submit a PR with logic changes and no test coverage

File: config/initializers/simple_form_bootstrap.rb (hunk 5 above)
Comment (josephferrairo) [inline config/initializers/simple_form_bootstrap.rb]: It'sfor styling, the readme states if you want to integrate simple_form with bootstrap to run the command rails g simple_form:install --bootstrap and that's what I did

File: config/database.yml (hunk 1 above)
Comment (josephferrairo) [inline config/database.yml]: fixed

**Comment by BGRicker** [comment]
This really needs to be at least two PRs. You want a PR to be a well organized and accomplishing a task, and I don't think there's any situation where you can consider it well organized when it's adding validations in the same PR as introducing testing tools like fabrication, guard, and shoulda etc.

**Pull request closed without merging**